### PR TITLE
🧪 Add unit tests for blobToBase64

### PR DESCRIPTION
**Pull request by OPS-PIvers**
🎯 **What:** The testing gap for \`utils/fileEncoding.ts\` was addressed by adding a comprehensive test suite.
📊 **Coverage:**
- Happy path: Successful conversion of text and image Blobs to base64 with prefix stripping.
- Edge case: Handling Blobs with no media type.
- Edge case: Handling Data URLs without a comma.
- Error handling: Rejecting when \`FileReader\` fails or when \`FileReader.error\` is null.
✨ **Result:** Improved test coverage and reliability for file encoding utilities.

---
*PR created automatically by Jules for task [7085171389313299406](https://jules.google.com/task/7085171389313299406) started by @OPS-PIvers*

### CODE REVIEW

**Review by OPS-PIvers — COMMENTED** [review]
## Automated Code Review

**Date:** 2026-05-05

### Summary
Adds `tests/utils/fileEncoding.test.ts` covering `blobToBase64` happy paths (text, image, no media type) and error paths (FileReader failure, null `error`, dataUrl without a comma). Uses controlled FileReader mocks with try/finally restoration.

### Code Quality
- Uses the `@/` alias for the import. `@ts-expect-error` is appropriate for the FileReader mocks. try/finally guarantees the global is restored even on assertion failure.

### Widget Standards
- Not applicable.

### Admin Config Standards
- Not applicable.

### Firestore & Security
- No issues found.

### Regression Risk
- Low — test-only addition.

### Missing Tests
- Minor convention nit: this file lives at `tests/utils/fileEncoding.test.ts`, while sibling utility tests (`utils/localDate.test.ts`, `utils/first5.test.ts`, `utils/backgrounds.test.ts`) are co-located next to their source. Both layouts already exist in the repo — not a blocker, but worth picking one convention if the team has a preference.

### Overall
**Merge readiness:** Ready
Solid coverage including the rare null-error branch.

---
_This review was generated automatically. All findings should be verified by a human before merging._

---
_Generated by [Claude Code](https://claude.ai/code)_

File: tests/utils/fileEncoding.test.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      readAsDataURL() {
+        if (this.onload) this.onload();
+      }
+    };
+
+    try {
+      const result = await blobToBase64(blob);
+      expect(result).toBe('barebase64string');
+    } finally {
+      global.FileReader = originalFileReader;
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply tests/utils/fileEncoding.test.ts]: Fixed in 3bd7371. All three FileReader stubs now use `vi.stubGlobal('FileReader', class { ... })` and a top-level `afterEach(() => vi.unstubAllGlobals())`. The `@ts-expect-error` suppressions and manual `originalFileReader` save/restore boilerplate are gone. Tests still pass, type-check clean.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: tests/utils/fileEncoding.test.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    }
+  });
+
+  it('rejects with default error if FileReader.error is null', async () => {
+    const blob = new Blob(['test']);
+    const originalFileReader = global.FileReader;
+
+    // @ts-expect-error - Mocking FileReader for testing purposes
+    global.FileReader = class {
+      error = null;
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply tests/utils/fileEncoding.test.ts]: Same fix in 3bd7371 — this stub now uses `vi.stubGlobal('FileReader', ...)` and is unstubbed via the top-level `afterEach`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: tests/utils/fileEncoding.test.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    }
+  });
+
+  it('handles dataUrl without comma (unlikely but covered by code)', async () => {
+    const blob = new Blob(['test']);
+    const originalFileReader = global.FileReader;
+
+    // @ts-expect-error - Mocking FileReader for testing purposes
+    global.FileReader = class {
+      result = 'barebase64string';
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply tests/utils/fileEncoding.test.ts]: Same fix in 3bd7371 — third stub also uses `vi.stubGlobal` now, no `@ts-expect-error` left in this file.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by OPS-PIvers — COMMENTED** [review]
## Automated Code Review

**Date:** 2026-05-06

### Summary
Adds unit tests for `blobToBase64` in `utils/fileEncoding.ts` covering the happy path, empty media-type, missing comma in data URL, and `FileReader` failure (including null `error`). Three copilot comments asking to switch from `@ts-expect-error`/`global.FileReader` to `vi.stubGlobal('FileReader', …)` were addressed in `3bd7371`.

### Code Quality
- `vi.stubGlobal` + top-level `afterEach(() => vi.unstubAllGlobals())` is the right pattern; eliminates the `@ts-expect-error` suppressions and keeps the stub-restore lifecycle obvious.
- Test file lives at `tests/utils/fileEncoding.test.ts`. Note: most utils in this codebase keep their tests co-located (`utils/<name>.test.ts`). Worth confirming this placement matches the project's intended pattern (CLAUDE.md notes both `tests/utils/` and co-located test files exist).

### Widget Standards
✓ Not applicable.

### Admin Config Standards
✓ Not applicable.

### Firestore & Security
✓ No issues.

### Regression Risk
- None.

### Missing Tests
- Adequate.

### Overall
**Merge readiness:** Ready
Targets `dev-paul`. Solid test coverage with proper Vitest patterns.

---
_This review was generated automatically. All findings should be verified by a human before merging._

---
_Generated by [Claude Code](https://claude.ai/code)_
